detectors/node/opentelemetry-resource-detector-container/src/detectors/ContainerDetector.ts

```diff
@@ -42,30 +44,56 @@ export class DockerCGroupV1Detector implements Detector {
           });
     } catch (e) {
       diag.info(
-        'Docker CGROUP V1 Detector did not identify running inside a supported docker container, no docker attributes will be added to resource: ',
+        'Container Detector did not identify running inside a supported container, no container attributes will be added to resource: ',
         e
       );
       return Resource.empty();
     }
   }
 
+  private async _getContainerIdV1() {
+    const rawData = await ContainerDetector.readFileAsync(
+      this.DEFAULT_CGROUP_V1_PATH,
+      this.UTF8_UNICODE
+    );
+    const splitData = rawData.trim().split('\n');
+    for (const str of splitData) {
+      if (str.length >= this.CONTAINER_ID_LENGTH) {
+        return str.substring(str.length - this.CONTAINER_ID_LENGTH);
+      }
+    }
+    return undefined;
+  }
+
+  private async _getContainerIdV2() {
+    const rawData = await ContainerDetector.readFileAsync(
+      this.DEFAULT_CGROUP_V2_PATH,
+      this.UTF8_UNICODE
+    );
+    const str = rawData
+      .trim()
+      .split('\n')
+      .find(s => s.includes(this.HOSTNAME));
+    const containerIdStr =
+      str && str.split('/').find(s => s.length >= this.CONTAINER_ID_LENGTH);
```

Comment:
nit: instead of `str && str.split(...` you can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to get the same effect in a cleaner syntax:
```js
const containerIdStr = str?.split('/').find(s => s.length >= this.CONTAINER_ID_LENGTH);
```

Reply:
done